src/code42cli/cmds/trustedactivities.py

```diff
@@ -0,0 +1,222 @@
+    "description",
+]
+TRUST_UPDATE_HEADERS = [
+    "resource_id",
+    "value",
+    "description",
+]
+TRUST_REMOVE_HEADERS = [
+    "resource_id",
+]
+
+trusted_activities_generate_template = generate_template_cmd_factory(
+    group_name="trusted_activities",
+    commands_dict={
+        "create": TRUST_CREATE_HEADERS,
+        "update": TRUST_UPDATE_HEADERS,
+        "remove": TRUST_REMOVE_HEADERS,
+    },
+    help_message="Generate the CSV template needed for bulk trusted-activities commands",
+)
+bulk.add_command(trusted_activities_generate_template)
+
+
+@bulk.command(
+    name="create",
+    help="Bulk create trusted activities using a CSV file with "
+    f"format: {','.join(TRUST_UPDATE_HEADERS)}.",
+)
+@read_csv_arg(headers=TRUST_CREATE_HEADERS)
+@sdk_options()
+def bulk_create(state, csv_rows):
+    """Bulk create trusted activities."""
+    sdk = state.sdk
+
+    def handle_row(type, value, description):
+        if type not in TrustedActivityType.choices():
+            message = f"Invalid type {type}, valid types include {', '.join(TrustedActivityType.choices())}."
+            raise Code42CLIError(message)
+        if type is None:
+            message = "'type' is a required field to create a trusted activity."
+            raise Code42CLIError(message)
+        if value is None:
+            message = "'value' is a required field to create a trusted activity."
+            raise Code42CLIError(message)
+        sdk.trustedactivities.create(type, value, description)
+
+    run_bulk_process(
+        handle_row, csv_rows, progress_label="Creating trusting activities:",
+    )
+
+
+@bulk.command(
+    name="update",
+    help="Bulk update trusted activities using a CSV file with "
+    f"format: {','.join(TRUST_UPDATE_HEADERS)}.",
+)
+@read_csv_arg(headers=TRUST_UPDATE_HEADERS)
+@sdk_options()
+def bulk_update(state, csv_rows):
+    """Bulk update trusted activities."""
+    sdk = state.sdk
+
+    def handle_row(resource_id, value, description):
+        if resource_id is None:
+            message = "'resource_id' is a required field to update a trusted activity."
+            raise Code42CLIError(message)
+        try:
+            _check_resource_id_type(resource_id)
+        except Code42CLIError:
+            raise
+
+        sdk.trustedactivities.update(resource_id, value, description)
+
+    run_bulk_process(
+        handle_row, csv_rows, progress_label="Updating trusted activities:"
+    )
+
+
+@bulk.command(
+    name="remove",
+    help="Bulk remove trusted activities using a CSV file with "
+    f"format: {','.join(TRUST_REMOVE_HEADERS)}.",
+)
+@read_csv_arg(headers=TRUST_REMOVE_HEADERS)
+@sdk_options()
+def bulk_remove(state, csv_rows):
+    """Bulk remove trusted activities."""
+    sdk = state.sdk
+
+    def handle_row(resource_id):
+        if resource_id is None:
+            message = "'resource_id' is a required field to remove a trusted activity."
+            raise Code42CLIError(message)
+        try:
```

Comment:
these try/except blocks aren't needed, you can just call `_check_resource_id_type`.